2024/articles/omp-santa.pod

```diff
@@ -0,0 +1,339 @@
+location one by one. We can use OpenMP to divide the problem into smaller
+parts, calculate distances in parallel, and get this fixed fast!”
+
+The data format of the positional satellites consisted of their positions in
+latitude, longitude, with an altitude; so the computations are necessarily in
+3D space, and could contain any number of lines:
+
+=begin data 
+
+  39.2497677581748 -66.1173923826129 29161.8658117126
+  -39.9677413540029 -41.3796046007432 23577.9949741844
+  82.2366387689737 153.417562140013 20022.1066827945
+  -43.4383552881127 -44.5406041422343 28011.9118605715
+  14.0767035175103 4.23608833137735 27766.7951824014
+  40.8573795733001 162.321349651587 25625.9162363042
+  -26.9656081428904 27.0935089406365 23681.3611776769
+  70.1644045619636 6.85910122836034 26946.7435635683
+  -64.5469805915126 -14.9091572762404 24893.2320145114
+  80.875127931392 -109.736894500449 23367.1123572306
+  5.62494420727084 -70.3599057022578 24677.4930437516
+  -78.594647140356 -69.1886836681495 21775.1041983417
+  -20.7093134304093 50.3824566178804 28396.5251214701
+  -8.19130244183 28.3379349990834 24113.3081697615
+  -62.1626942859846 -165.892484372947 27881.1415552865
+  -39.1505435434735 -14.0167682855066 25391.598017652
+  -14.701859640773 33.3797684668173 28958.4392020613
+  22.7094543766397 28.9295184727116 29847.2350918171
+  58.9987788505186 -87.6847921052664 29544.1317147911
+  83.6874173858257 -149.058764882263 22417.9224396509
+  -8.01336267852399 97.8876777856595 27879.4674084787
+  -55.3867650906297 -107.353651427755 21389.9702111095
+  -78.8588152992001 -155.558248147836 25430.0402744441
+  9.88995820014878 -0.204367766261981 20832.5618074863
+  -76.8565255868645 -14.4804333171123 27013.5287117141
+  -0.16890065869049 -40.7974093702016 22440.2960018416
+  8.56759320194605 14.0242190926548 24229.1350707098
+  89.3116725410715 19.3710347706399 28181.9446348641
+  ...
+
+=end data
+
+The distance formula was computationally expensive, involving square roots and
+trigonometry. But by using OpenMP, the task could be split into multiple OS
+threads, each calculating the distance to one satellite at the same time. The
+results would then be aggregated, and the sleigh's GPS would know exactly
+where to direct its heading.
+
+=begin c 
+
+  // Function to calculate Euclidean distance between two points (x1, y1, z1) and (x2, y2, z2)
+  double calculate_distance(double x1, double y1, double z1, double x2, double y2, double z2) {
+      return sqrt(pow(x2 - x1, 2) + pow(y2 - y1, 2) + pow(z2 - z1, 2));
+  }
+
+=end c
+
+The code would parallize the work by dividing the list of satellites
+among a number of I<pthreads> using OpenMP's C<#pragma omp for>
+construct! C<OpenMP::Simple> handles the required C<< #include <omp.h> >>
+and makes it easy to query the environment for information, such as C<OMP_NUM_THREADS>.
+
+=begin c 
+
+  PerlOMP_GETENV_BASIC // read common environmental variables, provided by OpenMP::Simple 
```

Comment:
Not totally sure what syntax is at play here. But shouldn't this have a semicolon at the end?